osu.Game/Screens/Play/PlayerSettings/BeatmapOffsetControl.cs

```diff
@@ -142,39 +146,42 @@ protected override void LoadComplete()
 
         private void currentChanged(ValueChangedEvent<double> offset)
         {
-            Scheduler.AddOnce(updateOffset);
+            // Negative is applied here because the play graph is considering a hit offset, not track (as we currently use for clocks).
+            lastPlayGraph?.UpdateOffset(-adjustmentSinceLastPlay);
+
+            // Calibration button may be hidden due to automatic offset adjustment, but it should be visible when the user manually adjusts their offset away from the applied suggestion.
+            calibrateFromLastPlayButton?.Show();
+
+            writeOffsetToBeatmap();
```

Comment:
```suggestion
            Scheduler.AddOnce(writeOffsetToBeatmap);
```

Else you could have two of these run at the same frame which is what we're trying to avoid.

Reply:
This **needs to be mentioned inline**.

I would change this without a blink of an eye because it currently goes against standards without any visible reason.